cmd/maintenance.go

```diff
@@ -83,17 +86,30 @@ func (c *controller) runMaintenance(cmd *cobra.Command, _ []string) error {
 		}
 		return pg.DoMaintenance(cmd.Context())
 	case redis:
-		r := maintenance.NewRedis(kubeClient, http.DefaultClient)
+		r := maintenance.NewRedis(kubeClient, getHTTPClient())
 		return r.DoMaintenance(cmd.Context())
 
 	case minio:
-		m := maintenance.NewMinio(kubeClient, http.DefaultClient)
+		m := maintenance.NewMinio(kubeClient, getHTTPClient())
 		return m.DoMaintenance(cmd.Context())
 
 	case mariadb:
-		m := maintenance.NewMariaDB(kubeClient, http.DefaultClient)
+		m := maintenance.NewMariaDB(kubeClient, getHTTPClient())
 		return m.DoMaintenance(cmd.Context())
+
+	case keycloak:
+		k := maintenance.NewKeycloak(kubeClient, getHTTPClient())
+		return k.DoMaintenance(cmd.Context())
 	}
 
 	return nil
 }
+
+func getHTTPClient() *http.Client {
+	if viper.GetString("REGISTRY_USERNAME") == "" || viper.GetString("REGISTRY_PASSWORD") == "" {
+		return &http.Client{
+			Timeout: 30 * time.Second,
+		}
+	}
+	return auth.GetAuthHTTPClient(viper.GetString("REGISTRY_USERNAME"), viper.GetString("REGISTRY_PASSWORD"))
+}
```

Comment:
shouldn't that be rather a secret?

Reply:
What do you mean by secret? The credentials are passed to the maintenance pod via env vars, we're reading them here again.

> so we allow just one image version? that won't work - we have request for different minor images right now, we can expect them later, maybe leave default and delete Enum?

Inventage currently only provides this image to us. So yes we only have this one at the moment. It can be extended in the future.